src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/GeneralizedCoastlineCheck.java

```diff
@@ -1,4 +1,117 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(GeneralizedCoastlineCheck.class);
+
+    public GeneralizedCoastlineCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.percentageThreshold = this.configurationValue(configuration, "thresholdPercentage",
+                MINIMUM_NODE_PAIR_THRESHOLD_PERCENTAGE);
+        this.minimumDistanceBetweenNodes = this.configurationValue(configuration,
+                "distanceBetweenNodes.minimum", MINIMUM_DISTANCE_BETWEEN_NODES);
+    }
+
+    /**
+     * This method validates or invalidates the supplied atlas object for the check
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked
+     * @return True if the {@link AtlasObject} is a {@link LineItem} with the tag natural=coastline,
+     *         or has a parent relation with that tag; false otherwise
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof LineItem
+                && (Validators.isOfType(object, NaturalTag.class, NaturalTag.COASTLINE)
+                        || ((LineItem) object).relations().stream()
+                                .anyMatch((Relation relation) -> Validators.isOfType(relation,
+                                        NaturalTag.class, NaturalTag.COASTLINE)));
+    }
+
+    /**
+     * This method flags appropriate atlas objects
+     *
+     * @param object
+     *            The {@link AtlasObject} being analyzed for potential flagging
+     * @return A flag on the parameter {@link AtlasObject} if it's a generalized coastline; an empty
+     *         Optional otherwise
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final double generalizedSegments = percentCoastlineSegmentsThatAreGeneralized(
+                (LineItem) object);
+        if (generalizedSegments >= this.percentageThreshold)
+        {
+            return Optional.of(this.createFlag(object, this.getLocalizedInstruction(0,
+                    generalizedSegments, this.minimumDistanceBetweenNodes)));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * This method calculates the percentage of {@link Segment}s in the {@link LineItem} whose
+     * lengths are greater than or equal to the minimumDistanceBetweenNodes configuration value
+     *
+     * @param line
+     *            The LineItem whose segments are evaluated
+     * @return The percentage of generalized {@link Segment}s in the parameter {@link LineItem},
+     *         0-100 inclusive.
+     */
+    private double percentCoastlineSegmentsThatAreGeneralized(final LineItem line)
+    {
+        int innerCount = 0;
+        final List<Segment> segments = line.asPolyLine().segments();
+        for (final Segment segment : segments)
+        {
+            if (segment.length().asMeters() >= this.minimumDistanceBetweenNodes)
```

Comment:
This could be done using the Distance class if you make `minimumDistanceBetweenNodes` a Distance object:
`segment.length().isGreaterThanOrEqualTo(this.minimumDistanceBetweenNodes)`

Reply:
Yes, but by keeping it as a primitive type I'd argue that a bit more memory is saved and at little to no cost of readability